### ISSUE TITLE
Are people actively using preprocess.default.py? There appears to be several bugs

### ISSUE THREAD

**Issue by Ben-J-Eppinger** [adjtomo/seisflows#173]
I am trying to switch form my stable fork of seisflows to seisflows 3. I work at exploration scales so usually just need to use the options provided in the default pre-processing module. However, I feel like there are several bugs in this implementation. For example, the muting functions don't appear to be implemented correctly because the source and receiver coordinates appear set to the same variable (the src_cords = get_reciever_cods(st)). That is just one example, I have found and fixed several other bugs and continue to find more.  I am asking because I am not sure if I am making silly mistakes, or trying to use functionally that is ostensibly extinct/unused

**Comment by bch0w** [comment]
Hi @Ben-J-Eppinger, thanks for raising this issue. 

You are definitely not making silly mistakes, and apologies if you've run into any issues using the default preprocessing module. I do use the default preprocessing module, but only the filtering portion. Many of the functions like muting and tapering that are useful at the exploration scale were rewritten from the original code, but have not been tested or used extensively, which is why there are likely several bugs floating around. 

If you would like to open up a pull request to submit bugfixes that you find, we would really appreciate it! Alternatively, if you point out any bugs you find here in this issue, I'd be happy to go in and fix those as soon as possible to get this module running smoothly!

**Comment by Ben-J-Eppinger** [comment]
Hi Bryant, thanks for getting back to me so quickly! I would be happy to submit some bug fixes to the repository, but upon further inspection, more of my problems seem to be coming from the fact that I am trying to work in the SU format. For example, the _check_adjoint_traces function works for the ASCII files specfem generates but not the SU ones. More importantly, the method of writing separate adjoint source files for each station rather than having one for each event seems to not be what specfem is expecting when using the SU format. I have been able to work past several of these issues, however, I am sure I have managed to do so in a way that doesn't break other functionality of the code, so I am not quite sure how best to proceed. I work in the near-surface, so I wonder if my use case is pretty outlandish compared to what most people are doing with Seisflows these days.

**Comment by bch0w** [comment]
Hi @Ben-J-Eppinger, just wanted to let you know I've seen this message and your post on the issue regarding the multi-core workflow, but I'll be out in the field the next few days so won't be able to get back to you until next week. I'll try to respond to you asap!

**Comment by Ben-J-Eppinger** [comment]
No worries, I really appreciate the help! This issue really has me stumped
so I am trying/looking forward to learning more.

**Comment by bch0w** [comment]
Hi @Ben-J-Eppinger, back in the office and working on SeisFlows again. 

I would say your use case is exactly what SeisFlows should be used for! And it's partly on me for dropping the ball on  carrying over support for things like SU format, which were available in the original codebase.

I'd be excited to ensure that you can use the latest codebase for your own work. Perhaps a good plan of action is to for me to get SU format support working, and then we can identify functionality that still requires some work, such as muting?

It would be great to work together on this as I am not very familiar with common preprocessing techniques required at exploration scales.

**Comment by bch0w** [comment]
Thanks @Ben-J-Eppinger, that's a great summary of the issue and will help with writing in an SU format reader/writer. The nice thing about SeisFlows is it's modularity, which allows us to easily write different interfaces for the 2D and 3D codes.

I think it is worthwhile to provide SU support, since it's useful for exploration scale problems, and will make SeisFlows stronger overall. In parallel I/we can work on getting those preprocessing functions working again, which should be straightforward bugfixing since the code is (mostly) written. 

Let me start a new issue and try to outline the required tasks for a pull request that does a bit of overhaul on the preprocessing module. I will try to chip away at this over the next few weeks!

**Comment by bch0w** [comment]
Hi @Ben-J-Eppinger, finally finding time to get back to old issues and this is a big one to fix. I'm closing this issue in favor of #174 and hopefully will make some progress on this soon. Thanks again for the outline in the above comments.
